src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/VideoBooking.kt

```diff
@@ -41,15 +47,58 @@ class VideoBooking private constructor(
 
   val createdBy: String,
 
-  val createdTime: LocalDateTime = LocalDateTime.now(),
+  val createdTime: LocalDateTime = now(),
 ) {
 
+  @OneToMany(mappedBy = "videoBooking", fetch = FetchType.LAZY, cascade = [CascadeType.ALL], orphanRemoval = true)
+  private val prisonAppointments: MutableList<PrisonAppointment> = mutableListOf()
```

Comment:
I have been moving/thinking towards managing the addition of appointments to the booking.  We can still have logic in the service but having these here helps with the cancellation journey.

Reply:
Is it the thin edge of the wedge which takes us down the road of the big entity graph, which turned into a big issue in A&A? We currently deal with prison appointments separately to the booking - not linked entities so won't this require a reasonably large refactor? When we add booking history rows there is another potential list of related entities in a many to one with the booking? I would rather every pull of a booking did not bring back its entire history. Generally, I think I prefer views to get the data we need, and for updates / cancellations to be specific by ID - but I do see the logic for appointments and how much clearer the cancel code is. What we don't want is the UI code walking through large entity graphs though. 50/50 for me...

Reply:
Think its ok ... provided we link entities very carefully, and only where we know there is no potential for growing data sets - like history rows.

Reply:
And I think this is fine for CRUD operations which are ID based.  Retrieval is via views.

Not doing this makes the service layer overly complicated e.g. more dependencies meaning multiple repos to create a booking when it can be done with one.  We know what we did in A and A so I think this risk can be avoided.

We were also dealing with much larger data sets in A & A.